### PR TITLE
0220 assign policy

### PR DESCRIPTION
**Pull request by mumian**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by mumian** [comment]
@bmoore-msft - Hi Brian, I need some guidance on how to by-pass this authorization issue.

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-azurepolicy-assign-buildinpolicy-resourcegroup%2Fazuredeploy.json" target="_blank">
```
Comment (bmoore-msft) [inline 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md]: Add a title to the readme.md

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-azurepolicy-assign-buildinpolicy-resourcegroup%2Fazuredeploy.json" target="_blank">
+  <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template assigns a build-in policy to an existing resource group. Inputs to this template are following fields:
```
Comment (bmoore-msft) [inline 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md]: build s/b built

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+- policyDefinitionID
+- policyAssignmentName
+
+The following PowerShell script shows how to get the policy definition ID and the policy display name of a built-in policy called "Audit resource location matches resource group location". The default policyAssignmentName is the policy display name.
+
+    PS C:\> $definition = Get-AzPolicyDefinition | Where-Object { $_.Properties.DisplayName -eq 'Audit resource location matches resource group location' }
+            $policyAssignmentName = $definition.Properties.displayName
```
Comment (bmoore-msft) [inline 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md]: remove these 2 lines

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md (hunk 2 above)
Comment (bmoore-msft) [inline 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md]: add a note that mentions you *must* be an owner of the subscription to apply a policy at this scope.

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/metadata.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "type": "QuickStart",
+  "itemDisplayName": "Assign a built-in policy to an existing Resource Group",
+  "description": "This template assigns a build-ii policy to an existing resource group.",
```
Comment (bmoore-msft) [inline 101-azurepolicy-assign-buildinpolicy-resourcegroup/metadata.json]: typo

built-in

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  "parameters": {
+    "policyAssignmentName": {
+      "type": "string",
+      "metadata": {
+        "description": "Specifies the name of the policy assignment. The default name is the built-in policy display name."
+      }
+    },
+    "rgName":{
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
```
[the remark this answers is not in the document]
Comment (mumian) [reply 101-azurepolicy-assign-buildinpolicy-resourcegroup/azuredeploy.json]: fixed

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "policyAssignmentName": {
+      "type": "string",
+      "metadata": {
+        "description": "Specifies the name of the policy assignment. The default name is the built-in policy display name."
```
[the remark this answers is not in the document]
Comment (mumian) [reply 101-azurepolicy-assign-buildinpolicy-resourcegroup/azuredeploy.json]: fixed

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md (hunk 1 above)
Comment (mumian) [reply 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md]: fixed.

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md (hunk 2 above)
Comment (mumian) [reply 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md]: fixed

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md (hunk 3 above)
Comment (mumian) [reply 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md]: @bmoore-msft - do you mean removing "$policyAssignmentName = $definition.Properties.displayName" or the whole powershell script?

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md (hunk 2 above)
Comment (mumian) [reply 101-azurepolicy-assign-buildinpolicy-resourcegroup/README.md]: Fixed.

File: 101-azurepolicy-assign-buildinpolicy-resourcegroup/metadata.json (hunk 4 above)
Comment (mumian) [reply 101-azurepolicy-assign-buildinpolicy-resourcegroup/metadata.json]: fixed

**Pull request merged**